tunnel/tunnel.go

```diff
@@ -17,21 +17,56 @@
 package tunnel
 
 import (
+	"encoding/json"
+	"time"
+
+	//"github.com/openziti/edge/tunnel/entities"
 	"github.com/openziti/sdk-golang/ziti/edge"
 	"github.com/sirupsen/logrus"
 	"io"
 	"net"
+	"strings"
 )
 
 var log = logrus.StandardLogger()
 
-func DialAndRun(provider FabricProvider, service string, clientConn net.Conn, halfClose bool) {
-	if err := provider.TunnelService(clientConn, service, halfClose); err != nil {
-		log.WithError(err).WithField("service", service).Error("tunnel failed")
+type Service interface {
+	GetName() string
+	GetDialTimeout() time.Duration
+}
+
+func DialAndRun(provider FabricProvider, service Service, clientConn net.Conn, appInfo map[string]string, halfClose bool) {
+	appInfoJson, err := json.Marshal(appInfo)
+	if err != nil {
+		log.WithError(err).WithField("service", service.GetName()).Error("unable to marshal appInfo")
+		_ = clientConn.Close()
+		return
+	}
+
+	if err := provider.TunnelService(service, clientConn, halfClose, appInfoJson); err != nil {
+		log.WithError(err).WithField("service", service.GetName()).Error("tunnel failed")
 		_ = clientConn.Close()
 	}
 }
 
+func GetIpAndPort(ipPort string) (string, string) {
```

Comment:
I would think this wants to handle strings that only contain an IP.

Reply:
It's only being called with the results of net.Addr.String(), which AFAIK will always have a host port. Maybe should convert it to take net.Addr directly...

Reply:
Updated it to take net.Addr directly and try type casting first